src/main/java/com/foodybuddy/dao/impl/ApartmentDAOImpl.java

```diff
@@ -8,38 +8,94 @@
 import com.foodybuddy.dao.ApartmentDAO;
 import com.foodybuddy.model.Apartment;
 
-
+/***
+ * Interface implementation for accessing Apartment Table
+ */
 @Repository
 public class ApartmentDAOImpl implements ApartmentDAO {
 	
 	private Session session = null;
 	
-	public ApartmentDAOImpl(Session session){
+	/***
+	 * Constructor
+	 * @param session
+	 * @throws NullPointerException
+	 */
+	public ApartmentDAOImpl(Session session) throws NullPointerException{
+	    if(session == null) {
+	        throw new NullPointerException("session passed to Apartment constructor is null");
+	    }
 		this.session = session;
 	}
-
-	public void insert(Apartment apartment) {
+	
+	/***
+	 * Apartment insert
+	 * @param apartment
+	 * @throws NullPointerException
+	 * @return apartment 
+	 */
+	public Apartment insert(Apartment apartment) throws NullPointerException {
+	    if(apartment == null) {
+            throw new NullPointerException("null apartment is passed to Apartment insert");
+        }
 		session.persist(apartment);
+		return apartment;
 	}
-
-	public void update(Apartment apartment) {
+	
+	/***
+     * Apartment update
+     * @param apartment
+     * @throws NullPointerException
+     * @return apartment 
+     */
+	public Apartment update(Apartment apartment) throws NullPointerException {
+	    if(apartment == null) {
+            throw new NullPointerException("null apartment is passed to Apartment update");
+        }
 		session.update(apartment);
+		return apartment;
 	}
+	
 
+	/***
+     * Apartment delete
+     * @param apartment
+     * @throws NullPointerException
+     * @return apartment 
+     */
+	/*
+	public Apartment delete(Apartment apartment) throws NullPointerException {
+	    if(apartment == null) {
+            throw new NullPointerException("null apartment is passed to Apartment delete");
+        }
+        session.delete(apartment);
+        return apartment;
+	}
+	*/
+	
+	/***
+     * Apartment getAll
+     * @return List<Apartment> : list of all apartments 
+     */
 	@SuppressWarnings("unchecked")
 	public List<Apartment> getAll() {
 		List<Apartment> ApartmentsList = session.createQuery("FROM Apartment").list();
 		return ApartmentsList;
 	}
-
-	public Apartment getById(int id) {		
+	
+	/***
+     * Apartment getById
+     * @param id
+     * @throws NullPointerException
+     * @return apartment 
+     */
+	public Apartment getById(Integer id) throws NullPointerException {
+	    if(id == null) {
+            throw new NullPointerException("null id is passed to Apartment getById");
+        }
 		String query = "FROM Apartment WHERE id = " + id;
 		Apartment apartment = (Apartment)session.createQuery(query).uniqueResult();
 		return apartment;
 	}
 
```

Comment:
What is the reason for removing delete function? for instance How will cleanup be done during testing